rpc.go

```diff
@@ -86,88 +86,81 @@ func (c *client) SendRPC(rpc hrpc.Call) (msg proto.Message, err error) {
 		sp.End()
 	}()
 
-	reg, err := c.getRegionForRpc(ctx, rpc)
-	if err != nil {
-		return nil, err
-	}
-
 	backoff := backoffStart
 	for {
-		msg, err = c.sendRPCToRegion(ctx, rpc, reg)
+		rc, err := c.getRegionAndClientForRPC(ctx, rpc)
+		if err != nil {
+			return nil, err
+		}
+		msg, err = c.sendRPCToRegionClient(ctx, rpc, rc)
 		switch err.(type) {
 		case region.RetryableError:
 			sp.AddEvent("retrySleep")
 			backoff, err = sleepAndIncreaseBackoff(ctx, backoff)
 			if err != nil {
 				return msg, err
 			}
+			continue // retry
```

Comment:
I can't imagine why it could be, it's just going to look up the region in the region cache again. And in the case the region has been split or moved since we started the backoff sleep on line 99 we will find the new region in fewer runs through this loop. Perhaps you have a different thought here given that you have dealt more with HBase issues?

I opted for this because it simplifies this code: One fewer error handling path to consider.

Reply:
Not really. Let's go with it and we will see :)